### DISCUSSION TITLE
Feature request: EmulationStation compatibility

### DISCUSSION THREAD

**Discussion by julianoBRL** [JMBeresford/retrom#212] [Ideas]
Hello, 

I have some arcade machines using RetroPie(EmulationStation+RetroArch) and i would like to know it there is any plans to make Retrom compatible with them, fo example the folders where the games are and the gamedata.xml.
I would love to use Retrom as a core for my arcade machines where i would mount the libraries on them and use Retrom as a central to download/add/update games acorss all my machines.

yours sincerely,
Juliano Lira

**Comment by JMBeresford** [comment]
I'm not opposed to eventually supporting this, can you provide some documentation for the file spec you mentioned?

**Comment by julianoBRL** [comment]
- Gamelists.xml:
https://github.com/Aloshi/EmulationStation/blob/master/GAMELISTS.md
This one is used to map the image, name, rating, room_path and others infos for each game. **/etc/emulationstation/gamelists/[SYSTEM_NAME]/gamelist.xml**

- Systems.xml
https://github.com/Aloshi/EmulationStation/blob/master/SYSTEMS.md
This one maps each available system with **path** of where the room folder are, extension to recognize rooms, and command to start the room.
This one stays in the emulation station root: **/etc/emulationstation**

- Themes.xml
i daubt you will need this one but it might be usefull:
https://github.com/Aloshi/EmulationStation/blob/master/THEMES.md
This one can stay on 2 folders:
**/etc/emulationstation/gamelists/[SYSTEM_NAME]/theme.xml**
**/etc/emulationstation/themes/[THEME_NAME]/[SYSTEM_NAME]/theme.xml**

Deafult folders that EmulationStation generates:
https://www.reddit.com/r/EmuDeck/comments/xb43rf/comment/inxbewn/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

The most important ones are the Systems.xml and Gamelists.xml for the EmulationStation.
The Systems.xml and  Gamelists.xml <path> it would help a lot the option for a custom path in cases like mine where the rooms folder will be shared across the network.
I'm not sure if the relative path or absolute path to use in the <path> tags.
The Themes.xml can be manually added later.

With this i can mount the Retrom library folder in N machines and have Retron manage all of them, including game data like images, rating, names, descriptions, developers and publishers.

Generating something like:
![image](https://github.com/user-attachments/assets/7742d747-4a11-4bf7-abef-2bc692f376f3)

**Comment by JMBeresford** [comment]
@julianoBRL Could you explain the nature of distribution that you are expecting to see here? Retrom currently expects _all_ game metadata to be owned by the server.

Even for third party integrations such as Steam, Retrom stores a user's library information on the server -- the client then is only responsible for tracking installation status and installing/launching the game via steam.

Are you intending to:

- use the Retrom server to manage/ingest your existing ES library
- have a Retrom client on each of your machines to manage _only installation/downloading of games_ and continue to use EmulationStation as your frontend/launcher?
	- If so, why not simply use the Retrom client in fullscreen mode as your frontend? I understand if you have a preference for ES, but then using Retrom underneath seems like it would be cumbersome

**Comment by julianoBRL** [comment]
I want to use Retrom as a central server to sync multiple Arcade Machines like in the image bellow, Retron would be a server to download rooms and generate/update the System.xml and Gamelists.xml and the arcade machines would have EmulationStation+Retroarch using the rooms downloaded from the Retrom and the System.xml and Gamelists also generated by the Retrom.

That way all machines would have the same rooms and settings without the need to download and update the Systems.xml and Gamelists.xml in each machine.

![image](https://github.com/user-attachments/assets/fd3c4ca2-cf7e-43b2-9c7a-247998b32bcd)

**Comment by JMBeresford** [comment]
I see now, this is unfortunately not something I would like Retrom to be responsible for.

Generating xml's for use in third party front-ends does not make sense to me, as it would essentially be duplicating the information that Retrom already manages in its own database. The behavior you describe is exactly what Retrom is designed for, however using the server currently requires using the Retrom client ( or at least using the server's API, see below ).

**Shimming support for other client frameworks/APIs into the Retrom server is simply not maintainable**. It would make much more sense for those other clients to instead have a plugin ( or similar ) that allows it to use the Retrom server. The server has a simple gRPC API that could easily be used by third-party clients that want to support it.

For your current use case, I can only recommend using the Retrom client instead of ES if you want to leverage the Retrom server. I am open to any feature requests for the client that would make that transition more acceptable!
